native/src/routes/SearchModal.tsx

```diff
@@ -126,77 +90,35 @@ class SearchModal extends React.Component<PropsType, SearchStateType> {
     closeModal(query)
   }
 
-  onSearchChanged = (query: string): void => {
-    this.setState({
-      query,
-    })
-  }
-
-  renderContent = (): ReactNode => {
-    const { language, cityCode, theme, categories } = this.props
-    const { query } = this.state
-    const minHeight = dimensions.categoryListItem.iconSize + dimensions.categoryListItem.margin * 2
-
-    if (!categories) {
-      return <ActivityIndicator size='large' color='#0000ff' />
-    }
-
-    const filteredCategories = this.findCategories(categories)
-    return (
-      <ScrollView
-        contentContainerStyle={{
-          flexGrow: 1,
-        }}
-        keyboardShouldPersistTaps='always'>
-        {/* The minHeight is needed to circumvent a bug that appears when there is only one search result.
-             See NATIVE-430 for reference. */}
-        <View
-          style={{
-            minHeight,
-          }}>
-          <CategoryList
-            categories={filteredCategories}
-            query={query}
-            onItemPress={this.onItemPress}
-            language={language}
-          />
-        </View>
-        {filteredCategories.length === 0 && <NothingFound />}
-        <FeedbackContainer
-          routeType={SEARCH_ROUTE}
-          isSearchFeedback
-          isPositiveFeedback={false}
-          language={language}
-          cityCode={cityCode}
-          query={query}
-          theme={theme}
-        />
-      </ScrollView>
-    )
-  }
-
-  render(): ReactNode {
-    const { theme, t } = this.props
-    const { query } = this.state
-    return (
-      <Wrapper theme={theme} {...testID('Search-Page')}>
-        <SearchHeader
-          theme={theme}
-          query={query}
-          closeSearchBar={this.onClose}
-          onSearchChanged={this.onSearchChanged}
-          t={t}
-        />
-        <KeyboardAvoidingView
-          behavior={Platform.OS === 'ios' ? 'padding' : undefined}
-          style={{
-            flex: 1,
-          }}>
-          {this.renderContent()}
-        </KeyboardAvoidingView>
-      </Wrapper>
-    )
-  }
+  const Content = searchResults ? (
+    <ScrollView contentContainerStyle={{ flexGrow: 1 }} keyboardShouldPersistTaps='always'>
+      {/* The minHeight is needed to circumvent a bug that appears when there is only one search result (NATIVE-430). */}
+      <View style={{ minHeight }}>
+        <CategoryList items={searchResults} query={query} onItemPress={onItemPress} language={language} />
+      </View>
+      {searchResults.length === 0 && <NothingFound />}
+      <FeedbackContainer
+        routeType={SEARCH_ROUTE}
+        isSearchFeedback
+        isPositiveFeedback={false}
+        language={language}
+        cityCode={cityCode}
+        query={query}
+        theme={theme}
+      />
+    </ScrollView>
+  ) : (
+    <ActivityIndicator size='large' color='#0000ff' />
```

Comment:
This `ActivityIndicator` feels really weird to me. It will only be shown when the `categoriesMapModel` is undefined, which should only happen when the `SearchRoute` is loading, right? To me the color is confusing, and why don't we use our own Loading spinner then?

Reply:
I'll replace it, was just that way before and didn't think about it.